Cwx.m

```diff
@@ -0,0 +1,360 @@
+// in any form;
+// - Software incoporating the K6TU code is provided free of charge to end users
+// - Source code of the package/software including the K6TU code must be Open Source
+// - Source code of the package/software including the k6TU code must be publicly
+//   available on the Internet via github or similar repository system
+//
+// Commercial Use
+// --------------
+// The incorporation of the K6TU software in a proprietary product regardless of
+// whether the product is sold for a fee, bundled with another product at no cost
+// or in any use by a for-profit organization is expressly prohibited without a
+// specific license agreement from Stu Phillips, K6TU and Ridgelift VC, LLC.
+//
+// Violation of these Copyright terms will be protected by US & International law.
+//
+
+#import <Foundation/Foundation.h>
+#import "Cwx.h"
+#import "Slice.h"
+
+#define MAX_CWX_DELAY_MS 2000
+#define MAX_CWX_SPEED 100
+#define MAX_NUMBER_OF_MACROS 12
+#define MIN_CWX_SPEED 5
+
+
+@interface Cwx()
+
+// Radio object to which this Cwx belongs
+@property (strong, nonatomic, readwrite) Radio *radio;
+// Pointer to private run queue for Cwx
+@property (strong, nonatomic) dispatch_queue_t cwxRunQueue;
+// Array of strings
+@property (nonatomic, readwrite) NSMutableArray *macros;
+
+@property (strong, nonatomic) NSDictionary *cwxTokens;
+
+
+@end
+
+
+enum cwxToken {
+    breakinDelay=0,
+    eraseSent,
+    macro,
+    sent,
+    wpm,
+};
+
+
+@implementation Cwx
+
+- (id)initWithRadio:(Radio *) radio {
+    self = [super init];
+    
+    self.radio = radio;
+    // populate the Macros array with empty strings
+    self.macros = [[NSMutableArray alloc] initWithCapacity:MAX_NUMBER_OF_MACROS];
+    for (int i=0 ; i < MAX_NUMBER_OF_MACROS ; i++) {
+        [_macros addObject: @""];
+    }
+    [self initcwxTokens];
+    
+    _cwxRunQueue = dispatch_queue_create("net.k6tu.cwx", DISPATCH_QUEUE_SERIAL);
+    
+    return self;
+}
+
+
+
+- (void) initcwxTokens {
+  self.cwxTokens = [[NSDictionary alloc] initWithObjectsAndKeys:
+                          [NSNumber numberWithInteger:breakinDelay], @"break_in_delay",
+                          [NSNumber numberWithInteger:eraseSent], @"erase",
+                          [NSNumber numberWithInteger:macro], @"macro",
+                          [NSNumber numberWithInteger:sent], @"sent",
+                          [NSNumber numberWithInteger:wpm], @"wpm",
+                          nil];
+}
+
+// Private macro to improve readibility of setters
+
+#define commandUpdateNotify(cmd, key, ivar, value) \
+/* Let observers know the change on the main queue */ \
+[self willChangeValueForKey:(key)]; \
+(ivar) = (value); \
+[self didChangeValueForKey:(key)]; \
+\
+__weak Cwx *safeSelf = self; \
+dispatch_async(self.cwxRunQueue, ^(void) { \
+/* Send the command to the radio on our private queue */ \
+[safeSelf.radio commandToRadio:(cmd)]; \
+});
+
+
+- (void) setDelay:(int)delay {
```

Comment:
Can't remember the original discussion but there was an issue on the Community about CWX and non-CWX delays being kept in sync - ie change one, and the other changes.

I don't know if this was done in the radio or in the client.

In SmartSDR, does changing the delay in CWX change the delay in the Cw panel?  If so, are status update messages sent from the radio ?